pkg/api/api.go

```diff
@@ -420,13 +421,15 @@ func (hs *HTTPServer) registerRoutes() {
 		// Validated query
 		apiRoute.Post("/dashboards/org/:orgId/uid/:dashboardUid/panels/:panelId/query", authorize(reqSignedIn, ac.EvalPermission(datasources.ActionQuery)), routing.Wrap(hs.QueryMetricsFromDashboard))
 
-		apiRoute.Group("/alerts", func(alertsRoute routing.RouteRegister) {
-			alertsRoute.Post("/test", routing.Wrap(hs.AlertTest))
-			alertsRoute.Post("/:alertId/pause", reqEditorRole, routing.Wrap(hs.PauseAlert))
-			alertsRoute.Get("/:alertId", hs.ValidateOrgAlert, routing.Wrap(hs.GetAlert))
-			alertsRoute.Get("/", routing.Wrap(hs.GetAlerts))
-			alertsRoute.Get("/states-for-dashboard", routing.Wrap(hs.GetAlertStatesForDashboard))
-		})
+		if !hs.Cfg.UnifiedAlerting.IsEnabled() && setting.AlertingEnabled != nil && *setting.AlertingEnabled {
```

Comment:
The legacy alerting service will not register if `!setting.ExecuteAlerts`, but this expression isn't checking that. Should it?

I am not sure what this setting does, so this comment might be naive. I am simply unsure why we would check for it in one place but not the other.

Reply:
Afaiu, ExecuteAlerts controls the engine (aka scheduler). If it is disabled the rule API was still available, which IMO does not make sense.